src/Event.php

```diff
@@ -81,6 +99,9 @@ public function getTarget()
      * Overwrites parameters
      *
      * @param  array|ArrayAccess|object $params
+     * @template NewTParams of \ArrayAccess|array
```

Comment:
The previous signature stated `array|ArrayAccess|object`, but here we use `ArrayAccess|array`? `object` was kinda lost :thinking: